dao/src/main/java/com/epam/esm/dao/specification/gift/GetGiftCertificatesByNameDescrTagsSpecification.java

```diff
@@ -0,0 +1,101 @@
+package com.epam.esm.dao.specification.gift;
+
+import com.epam.esm.dao.entity.GiftCertificate;
+import com.epam.esm.dao.entity.Tag;
+import com.epam.esm.dao.sort.GiftCertificateSorter;
+import com.epam.esm.dao.specification.Specification;
+
+import javax.persistence.criteria.CriteriaBuilder;
+import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Join;
+import javax.persistence.criteria.Path;
+import javax.persistence.criteria.Predicate;
+import javax.persistence.criteria.Root;
+import javax.persistence.criteria.Subquery;
+import java.util.ArrayList;
+import java.util.List;
+
+public final class GetGiftCertificatesByNameDescrTagsSpecification
```

Comment:
byNameAndDescr... 
The name is really unintuitive

Reply:
Thank you. I made up a more appropriate name for this class. (fixed)